lib/Sema/CSSimplify.cpp

```diff
@@ -11404,6 +11404,7 @@ ConstraintSystem::simplifyApplicableFnConstraint(
             FunctionType::get(trailingClosureTypes, callAsFunctionResultTy,
                               FunctionType::ExtInfo());
 
+        increaseScore(SK_DisfavoredOverload);
```

Comment:
Could there be any unexpected consequences of using `SK_DisfavoredOverload` here instead of a separate score kind? Would we want a plain call to have an increased score for anything above disfavored overload, e.g. a conversion to `Any`, to be used instead of `callAsFunction` without one? The answer might be yes, just want to make sure we think through it!

Reply:
This is a good question. My thinking is that `callAsFunction` injection makes the solution overall worse than anything else even if other overloads require conversions because such overloads are closest possible match to what user wrote so it would be least surprising for them to match.

Reply:
We talked offline about this and came to the conclusion that this is appropriate.